cmd/tracee-ebpf/capabilities.go

```diff
@@ -73,16 +74,47 @@ func getCapabilitiesRequiredByTraceeEvents(cfg *tracee.Config) []cap.Value {
 	return removeDupCaps(caps)
 }
 
+// Retrieve the value of the kernel parameter perf_event_paranoid
+func getKernelPerfEventParanoidValue() int {
+	// Read the kernel parameter perf_event_paranoid
+	value, err := os.ReadFile("/proc/sys/kernel/perf_event_paranoid")
+	// If file read failed, returns the most restrictive value
+	if err != nil {
+		return 4
+	}
+	// Strip the newline character
+	value = value[:len(value)-1]
+	// Convert the value to an integer
+	intVal, err := strconv.ParseInt(string(value), 0, 16)
+	// If conversion failed, returns the most restrictive value
+	if err != nil {
+		return 4
+	}
+	return int(intVal)
+}
+
 // Get all capabilities required for eBPF usage (including perf buffers maps management)
 func getCapabilitiesRequiredByEBPF(OSInfo KernelVersionInfo) ([]cap.Value, error) {
 	// In kernel 5.8, CAP_BPF and CAP_PERFMON capabilities were introduced in order to replace CAP_SYS_ADMIN when
 	// loading eBPF programs.
-	// For some reasons, some distributions using new kernels still need CAP_SYS_ADMIN,
-	// so tracee still use it instead of the new capabilities.
-	caps := []cap.Value{
-		cap.IPC_LOCK,
-		cap.SYS_RESOURCE,
-		cap.SYS_ADMIN,
+	kernelParanoidValue := getKernelPerfEventParanoidValue()
+	var caps []cap.Value
+	// if kernelParanoidValue is too high, CAP_SYS_ADMIN is required
+	if kernelParanoidValue > 1 {
+		fmt.Println("Kernel perf_event_paranoid value is greater than 1, thus Tracee needs to use the CAP_SYS_ADMIN capability instead of CAP_BPF + CAP_PERFMON.")
```

Comment:
We can't write messages to stdout in non fatal errors I'm afraid. We should write to stderr OR, even better, to stdout if debug flag is given.

Reply:
I think it should probably be printed if in debug mode.
I hope that in the near future we will have a logger to handle this logic, but you can just change it so you get the debug flag as a argument to this function.

Reply:
I took the liberty to add the config to the function parameters to check the Debug flag, let me know if I shouldn't.

Reply:
Hmm I think we should only pass the flag for now, to make it less dependent and more testable.